src/neml2/models/solid_mechanics/CubicElasticityTensor.cxx

```diff
@@ -0,0 +1,153 @@
+// By: Argonne National Laboratory
+// OPEN SOURCE LICENSE (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "neml2/models/solid_mechanics/CubicElasticityTensor.h"
+
+namespace neml2
+{
+register_NEML2_object(CubicElasticityTensor);
+
+OptionSet
+CubicElasticityTensor::expected_options()
+{
+  OptionSet options = ElasticityTensor::expected_options();
+  options.doc() += "  This class defines an isotropic elasticity tensor using two parameters."
+                   "  Various options are available for which two parameters to provide.";
+
+  options.set_parameter<CrossRef<Scalar>>("p1");
+  options.set("p1").doc() = "First parameter";
+
+  EnumSelection type_selection({"youngs_modulus", "poissons_ratio", "shear_modulus", "INVALID"},
+                               {static_cast<int>(CubicElasticityTensor::ParamType::YOUNGS),
+                                static_cast<int>(CubicElasticityTensor::ParamType::POISSONS),
+                                static_cast<int>(CubicElasticityTensor::ParamType::SHEAR),
+                                static_cast<int>(CubicElasticityTensor::ParamType::INVALID)},
+                               "INVALID");
+  options.set<EnumSelection>("p1_type") = type_selection;
+  options.set("p1_type").doc() =
+      "First parameter type. Options are: " + type_selection.candidates_str();
+
+  options.set_parameter<CrossRef<Scalar>>("p2");
+  options.set("p2").doc() = "Second parameter";
+
+  options.set<EnumSelection>("p2_type") = type_selection;
+  options.set("p2_type").doc() =
+      "Second parameter type. Options are: " + type_selection.candidates_str();
+
+  options.set_parameter<CrossRef<Scalar>>("p3");
+  options.set("p3").doc() = "Third parameter";
+
+  options.set<EnumSelection>("p3_type") = type_selection;
+  options.set("p3_type").doc() =
+      "Third parameter type. Options are: " + type_selection.candidates_str();
```

Comment:
It would be nicer if we ask for a list of parameters and a list of parameter types, so that the input file looks something like
```python
[Models]
  [C]
    type = CubicElasticityTensor
    param_types = 'YOUNGS POISSONS SHEAR'
    param_values = '1 0.2 3'
  []
[]
```